### ISSUE TITLE
Sensors are not updateing

### ISSUE THREAD

**Issue by barackca1** [laszlojakab/homeassistant-dijnet#71]
Hello! 

I use the newest HA (Home Assistant 2023.2.5, with Supervisor 2023.01.1) My dijnet sensors are not updateing, only when I restart my HA. I treid update with entity update service, but not updateing. If I try manual update then in the log (/var/log/syslog and messages too) I see this, what I cannot understand (what does ewelink have to do with dijnet):

Feb 21 20:38:41 (my-server-name) addon_f4f71350_**ewelink_smart_home**_slug[803]: 2023-02-21T19:38:41.322Z [INFO] HA emit call_service event: {"domain":"homeassistant","service":"update_entity","service_data":{"entity_id":["sensor.dijnet_mvm_next_foldgaz_fizetendo_osszeg","sensor.dijnet_mvm_next_foldgaz_fizetendo_osszeg_2"]}}

In HA log I see this (but this independent from the above): 

2023-02-21 19:51:25.914 WARNING (MainThread) [homeassistant.helpers.entity] Update of sensor.dijnet_dijbeszedo_zrt_fizetendo_osszeg is taking over 10 seconds

Thank you!

**Comment by cociweb** [comment]
Hello,
My sensors also not working, for weeks/months, but I did not recognized it for long time and I got some invoice overdue..

So here is my error from the log. currently I'm on >2023.06 so python 3.11 is used:

```
2023-06-11 00:44:30.009 ERROR (MainThread) [homeassistant.helpers.entity] Update for sensor.dijnet_fcsm_zrt_fizetendo_osszeg fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 559, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 784, in async_device_update
    await self.async_update()
  File "/config/custom_components/dijnet/sensor.py", line 121, in async_update
    invoices = [invoice for invoice in await self._controller.get_unpaid_invoices() if invoice.display_name == self._invoice_issuer.display_name and invoice.provider == self._provider]
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/dijnet/controller.py", line 362, in get_unpaid_invoices
    await self.update_invoices()
  File "/config/custom_components/dijnet/controller.py", line 443, in update_invoices
    self._initialize_registry_and_unpaid_invoices()
  File "/config/custom_components/dijnet/controller.py", line 679, in _initialize_registry_and_unpaid_invoices
    if isinstance(registry[ATTR_REGISTRY_NEXT_QUERY_DATE], datetime):
                  ~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable
```

**Comment by skurusa** [comment]
On 2024.01.23. 2:11pm was stopped, the payment goes to 0.

**Comment by laszlojakab** [comment]
Is this solves your problem? https://github.com/laszlojakab/homeassistant-dijnet/issues/76#issuecomment-1913719486

**Comment by skurusa** [comment]
Szia!Redownload majd restart után jelenleg még nem frissül.Minden marad 0Ft-on.Ami infó össz megjelenik logban:2024-01-28 22:47:00.142 INFO (MainThread) [custom_components.dijnet.sensor] Setting up Dijnet sensors.2024-01-28 22:47:00.142 DEBUG (MainThread) [custom_components.dijnet.controller] Updating issuers.2024-01-28 22:47:01.735 DEBUG (MainThread) [custom_components.dijnet.dijnet_session] Posting login information to login_check_ajax.2024-01-28 22:47:02.477 DEBUG (MainThread) [custom_components.dijnet.dijnet_session] Getting main page.2024-01-28 22:47:02.943 ERROR (Üdv.: SanyiSent from my iPhone2024. jan. 28. dátummal, 21:57 időpontban Laszlo Jakab ***@***.***> írta:﻿
Is this solves your problem? #76 (comment)

**Comment by cociweb** [comment]
Patch applied, waiting for the next invoice in díjnet...

**Comment by skurusa** [comment]
Szia!

[image: image.png]

Ez normális?

Üdv.: Sanyi

Laszlo Jakab ***@***.***> ezt írta (időpont: 2024. jan. 28.,
V, 21:57):

**Comment by cociweb** [comment]
Next invoice is arrived, and I can confirm that sensors are updated now! @laszlojakab Thank you for your work! Can you please release it officially? (I bet Sanyi just re-downloaded it from hacs without the patch, aren't you?)

**Comment by skurusa** [comment]
Yes, arrived.
Working fine!
Thank you!

**Comment by laszlojakab** [comment]
Fixed in #77
